reader/reader_test.go

```diff
@@ -0,0 +1,36 @@
+package reader
+
+import (
+	"log"
+	"os"
+	"testing"
+)
+
+func TestReadCsv(t *testing.T) {
+	name := "test.csv"
+	f := createFile(name)
+
+	dat := ReadCsv(name)
+	if len(dat) != 2 {
+		log.Println("Map size differs than expected in readfile")
+		t.Fail()
+	}
+
+	if dat["kOne"] != "valOne" && dat["kTwo"] != "valTwo" {
+		log.Println("Fail while reading file content")
+		t.Fail()
+	}
+	f.Close()
+}
+func createFile(name string) *os.File {
+	f, err := os.Create(name)
+	if err != nil {
+		panic(err)
+	}
+	f.WriteString("kOne,valOne,")
+	f.WriteString("\n")
+	f.WriteString("kTwo,valTwo,")
+	f.WriteString("\n")
+
+	return f
+}
```

Comment:
I have refactored your test. However this is not correctly targeting everything. For a project this size this way of writing test make total sense though.
```go
package reader_test // tests usually go into a separate package.

import (
	"bufio"
	"fmt"
	"io/ioutil"
	"os"
	"testing"

	"github.com/kumit12/quiz/reader"
)

type testCase struct {
	key, value string
}

func TestReadCsv(t *testing.T) {
	tcs := []testCase{ // now you can add as many values as you want
		{"kOne", "valOne"},
		{"kTwo", "valTwo"},
		{"PUYaCC", "PZLAcvM"},
		{"nhGtbwdtz", "WfaVY"},
		{"11", "RdMESmffxz"},
	}

	name, cleanUp, err := createFile(tcs)
	if err != nil {
		t.Fatalf("writing file: %v", err)
	}
	defer cleanUp()

	data := reader.ReadCsv(name)
	if len(data) != len(tcs) {
		t.Fatalf("len(data) = (%d); want (%d)", len(data), len(tcs))
	}

	for _, tc := range tcs {
		// note that data comes in key=>value format.
		if _, ok := data[tc.key]; !ok { // here we are checking for key existence
			t.Errorf("(%s) not in data keys", tc.key)
			continue // otherwise the next check will panic. also we can check the rest of the file
		}
		if data[tc.key] != tc.value { // we check the value separately because we want to know where it went wrong
			t.Errorf("data[tc.key] = (%s); want (%s)", data[tc.key], tc.value)
		}
	}
}

// createFile returns a filename, a clean up function which helps
// cleaning up the artefacts, and an error if any occurs.
func createFile(tcs []testCase) (string, func(), error) { // note that we are not failing here at all.
	f, err := ioutil.TempFile("", "quiz")
	if err != nil {
		return "", nil, err
	}
	defer f.Close()

	w := bufio.NewWriter(f) // this writer is more efficient (it's buffered)
	for _, tc := range tcs {
		fmt.Fprintf(w, "%s,%s\n", tc.key, tc.value)
	}

	err = w.Flush() // always remember to flush it.
	if err != nil {
		return "", nil, err
	}
	cleanUp := func() {
		err := os.Remove(f.Name()) // probably you should check the error, but there is nothing you can do about it at this point.
		if err != nil {
			panic(err)
		}
	}
	return f.Name(), cleanUp, nil
}
```